src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const data in action.extraData) {
+          state[data] = action.extraData[data];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const key in action.keysToRemove) {
```

Comment:
[CODE STYLE]: don't use for in loop for iterating over array
keysToRemove is an array, please don't use for in loop here as it says in the checklist. I suggest using for of loop